### PR TITLE
feat: gradient clipping support for `LightningAdapter`

### PR DESCRIPTION
**Pull request by lqf96**
## Description
Add gradient clipping support for `LightningAdapter`. Parameter names are borrowed from the latest PyTorch Lightning [`Trainer`](https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer-class-api) class.

## Test Plan
(TBD; not sure where to add tests since this is my first contribution to the project)

## Commentary (optional)
Right now `_gradient_clip_val` and `_gradient_clip_algorithm` are stored within the `LightningAdapter`. Is this okay or should these be moved to other places? (e.g. `_LightningAdapterState`)

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by rb-determined-ai** [comment]
Hi @lqf96, we were on a company retreat yesterday, and I am pretty busy today but I will try and get this reviewed on monday.

In the mean time, thank you for your PR and your CLA looks good.  @cla-bot[bot] check

File: harness/determined/pytorch/lightning/_adapter.py [hunk 1]
```diff
@@ -131,6 +132,8 @@ def __init__(
         precision: Union[Literal[32], Literal[16]] = 32,
         amp_backend: Union[Literal["native"], Literal["apex"]] = "native",
         amp_level: Literal["O0", "O1", "O2", "O3"] = "O2",
+        gradient_clip_val: Union[int, float, None] = None,
+        gradient_clip_algorithm: Union[Literal["norm"], Literal["value"]] = "norm",
```
Comment (rb-determined-ai) [inline harness/determined/pytorch/lightning/_adapter.py:136]: Asking users to pass these values here is a functional solution, meaning that users can use it and it will work.

But it is not a very "PytorchLightning-native" solution.  It would be really nice if we could somehow hook directly into the `configure_gradient_clipping` call directly on the LightningModule that the user sets up.

I don't think this would be terribly hard either, though from looking at LightningModule.clip_gradients, we'd have to makes sure to patch clip_gradients() to not try and read `self.trainer`.  I think it would be a very nice user experience though.

What do you think?

**Comment by rb-determined-ai** [comment]
closing due to inactivity.  Happy to reopen later if @lqf96 is still interested in this work.
